apps/mark-scan/backend/src/pat-input/patinputd.c

```diff
@@ -0,0 +1,170 @@
+#include <fcntl.h>
+#include <linux/uinput.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include <time.h>
+#include <signal.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <string.h>
+
+#include "vsapgpio.h"
+
+bool should_exit_cleanly = false;
+char *pat_is_connected_gpio_number = "478";
+char *pat_a_signal_gpio_number = "481";
+char *pat_b_signal_gpio_number = "476";
+
+const long INTERVAL_MS = 100;
+struct timespec interval_timespec = {
+    0,
+    (INTERVAL_MS % 1000) * 1000000,
+};
+struct timespec rem;
+
+void emit(int fd, int type, int code, int val)
+{
+  struct input_event ie;
+
+  ie.type = type;
+  ie.code = code;
+  ie.value = val;
+  /* timestamp values below are ignored */
+  ie.time.tv_sec = 0;
+  ie.time.tv_usec = 0;
+
+  write(fd, &ie, sizeof(ie));
+}
+
+void interrupt(int signal)
+{
+  printf("Got exit signal %d\n", signal);
+  should_exit_cleanly = true;
+}
+
+int main(void)
+{
+  struct uinput_setup usetup;
+
+  int uinput_fd = open("/dev/uinput", O_WRONLY | O_NONBLOCK);
+
+  /*
+   * The ioctls below will enable the device that is about to be
+   * created, to pass key events.
+   */
+  ioctl(uinput_fd, UI_SET_EVBIT, EV_KEY);
+  ioctl(uinput_fd, UI_SET_KEYBIT, KEY_1);
+  ioctl(uinput_fd, UI_SET_KEYBIT, KEY_2);
+
+  memset(&usetup, 0, sizeof(usetup));
+  usetup.id.bustype = BUS_USB;
+  usetup.id.vendor = 0x1234;  /* sample vendor */
+  usetup.id.product = 0x5678; /* sample product */
```

Comment:
Sample?

Reply:
We don't need to reference this device by vendor or product ID ever so these values don't matter. An official vendor ID is something you need to register with the USB Implementers Forum so afaik we don't have one.

Reply:
If it doesn't matter, do you still need to set it? If you do need to set it, maybe a comment explaining that the values don't matter would be helpful, and a little clearer than "sample".

Reply:
Good point - [e6c9729](https://github.com/votingworks/vxsuite/pull/4210/commits/e6c972970510cdadcccf9220a2c022dbb0f5d4ed)